provdebug/pvdebug.py

```diff
@@ -53,37 +58,68 @@ def run():
             - q or quit: quit the replayer
         ''')
         
-    parser = ArgumentParser(description="provdb starts a provenance-based time traveling debugging interface.",
+    argParser = ArgumentParser(description="provdb starts a provenance-based time traveling debugging interface.",
     formatter_class=RawDescriptionHelpFormatter,
     epilog=helpText)
     '''
     #TO DEBUG
-    parser.add_argument('--interactive', action='store_true', default=True)
-    (args, rest) = parser.parse_known_args()
+    argParser.add_argument('--interactive', action='store_true', default=True)
+    (args, rest) = argParser.parse_known_args()
     if args.interactive:
         try: readline.read_history_file()
         except: pass
         rest += input("Arguments: ").split(" ")  # get input args
         try: readline.write_history_file()
         except: pass
     '''
-    parser.add_argument("-f", "--file", dest="file", required=False,
+    argParser.add_argument("-f", "--file", dest="file", required=False,
                         help="Prov.Json file to analyze")
 
-    parser.add_argument("-r", "--replay", dest="file", required=False,
+    argParser.add_argument("-r", "--replay", dest="file", required=False,
                         help="Prov.Replay file to analyze")
 
+    argParser.add_argument("-d", "--diff", nargs='+', required=False,
+                        help="Prov.Json files to compare")
+
     #TO DEBUG
     #args = parser.parse_args(rest)
-    
-    args = parser.parse_args()
+
+    args = argParser.parse_args()
 
     provdb_command = sys.argv[1]
+    if provdb_command == "-d" or provdb_command == "-diff":
+        provided_files = args.diff
+        if not all(name.endswith(".json") for name in provided_files):
+            print("The following file extension for diff is required: .json")
+            return
+        print("Welcome to the Multilingual Provenance Debugger in Diff Mode, type help for more information.")
```

Comment:
I just realized: in our current workflow, we immediately print out the diff here, so maybe the part

> ... type help for more information.

isn't super accurate? This would change later on once we finalize how this is going to look to the user

Reply:
Good catch